bee-network/bee-autopeering/src/discovery/query.rs

```diff
@@ -0,0 +1,287 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    discovery::manager,
+    event::EventTx,
+    local::Local,
+    peer::{
+        peerlist::{ActivePeer, ActivePeersList, MasterPeersList, ReplacementList},
+        PeerId,
+    },
+    request::RequestManager,
+    server::ServerTx,
+    task::Repeat,
+};
+
+use rand::{thread_rng, Rng};
+
+#[derive(Clone)]
+pub(crate) struct QueryContext {
+    // TODO: revisit dead code
+    #[allow(dead_code)]
+    pub(crate) local: Local,
+    pub(crate) request_mngr: RequestManager,
+    pub(crate) master_peers: MasterPeersList,
+    pub(crate) active_peers: ActivePeersList,
+    pub(crate) replacements: ReplacementList,
+    pub(crate) server_tx: ServerTx,
+    pub(crate) event_tx: EventTx,
+}
+
+// Hive.go: pings the oldest active peer.
```

Comment:
I would keep them until we are confident enough (ran it for a while on the mainnet) about our impl/port. I added them so that I don't have to switch back and forth even after having not worked on it for weeks. At some point all those comments can go away, but I think they are fine for 0.1.0 :slightly_smiling_face: